consensus/src/processes/pruning_proof/mod.rs

```diff
@@ -672,30 +1037,35 @@ impl PruningProofManager {
                     block_at_depth_2m
                 };
 
+                // new root is expected to be always an ancestor of old_root because new root takes a safety margin
+                assert!(self.reachability_service.is_dag_ancestor_of(root, old_root));
+
                 let mut headers = Vec::with_capacity(2 * self.pruning_proof_m as usize);
                 let mut queue = BinaryHeap::<Reverse<SortableBlock>>::new();
                 let mut visited = BlockHashSet::new();
-                queue.push(Reverse(SortableBlock::new(root, self.ghostdag_stores[level].get_blue_work(root).unwrap())));
+                queue.push(Reverse(SortableBlock::new(root, self.headers_store.get_header(root).unwrap().blue_work)));
                 while let Some(current) = queue.pop() {
                     let current = current.0.hash;
                     if !visited.insert(current) {
                         continue;
                     }
 
-                    if !self.reachability_service.is_dag_ancestor_of(current, selected_tip) {
+                    if !self.reachability_service.is_dag_ancestor_of(current, selected_tip)
+                        || !ghostdag_stores[level].has(current).is_ok_and(|found| found)
```

Comment:
we are iterating from `root` up in the exact same way `fill_level_proof_ghostdag_data` iterates up when filing `ghostdag_stores[level]`. So it seems this `has` check will always return true

Reply:
I think it's important to retain this check even though it seems redundant since it is the only thing that ties the work done in `fill_level_proof_ghostdag_data` to the actual set of headers that goes into the proof (which is constructed here). It serves as an assertion that it must be the same headers that we arrive at.

Reply:
See the latest comment I added